ui/app/rif/components/modal/addNewTokenNetworkAddress/addNewTokenNetworkAddress.js

```diff
@@ -0,0 +1,90 @@
+import React, { Component } from 'react'
+import PropTypes from 'prop-types'
+import { connect } from 'react-redux'
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
+import Select from 'react-select'
+
+class AddNewTokenNetworkAddress extends Component {
+	static propTypes = {
+		updateNetwork: PropTypes.func.isRequired,
+		updateAddress: PropTypes.func.isRequired,
+		networks: PropTypes.array.isRequired,
+	}
+    constructor(props) {
+        super(props);
+		let networks = []
+		Object.assign(networks, props.networks)
+		this.state = { 
+            networks: networks,
+			selectedNetwork: networks[0],
+			insertedAddress: '',
+		};
+	}
+	_updateNetwork = (selectedOption) => {
+		this.setState({ selectedNetwork: selectedOption })
+		this.props.updateNetwork(selectedOption)
+	}
+	_updateAddress = (e) => {
+		this.setState({ insertedAddress: e.target.value })
+		this.props.updateAddress(e.target.value)
+	}
+	
+	render () {
+        const {  } = this.props
+        const selectValue = ({value}) => {
+			return(
+			<div className={'add-new-multicrypto-select-value'}>
+				<span>
+				<FontAwesomeIcon  className={'add-new-multicrypto-select-value-icon'} icon={value.icon} color={value.color}/>
+					<span className={'add-new-multicrypto-select-value-text'}>{value.label}</span>
+				</span>
+			</div>
+		)}
+		const selectOption = (props) => {
+			let option = props.option
+			return(
+				<div
+					onMouseDown={(event) => {
+						event.preventDefault();
+						event.stopPropagation();	
+						props.onSelect(props.option, event)
+					}}
+					onMouseEnter={(event) => props.onFocus(props.option, event)}
+					onMouseMove={(event) => {
+							if (props.isFocused) return;
+							props.onFocus(props.option, event)
+					}}
+				>
+					<FontAwesomeIcon className={'add-new-multicrypto-select-value-icon'} icon={option.icon} color={option.color}/>
+					<span>&nbsp;{option.label}</span>
```

Comment:
I recommend a css margin-left here instead, just thinking in the future, maybe we will need to have more or less spacing and this solution is not the best on that case.

Reply:
Agree

Reply:
Ok, i added a new className in a css for common usage.